github_scripts/version_test.sh

```diff
@@ -0,0 +1,84 @@
+#!/bin/bash -xe
+
+# Copyright (C) 2024, Pelican Project, Morgridge Institute for Research
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you
+# may not use this file except in compliance with the License.  You may
+# obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+# This tests the --version flag of various Pelican binaries (pelican/stash/osdf)
+
+set -e
+
+mkdir -p /tmp/pelican-test/version-test
+
+export PELICAN_CONFIGDIR=/tmp/pelican-test/version-test
+export OSDF_CONFIGDIR=/tmp/pelican-test/version-test
+export STASH_CONFIGDIR=/tmp/pelican-test/version-test
```

Comment:
With the merge of #1162, let's avoid setting the now-extraneous OSDF/STASH env vars